security-jwt/src/main/java/io/micronaut/security/token/jwt/signature/jwks/ResourceRetrieverJwksClient.java

```diff
@@ -31,22 +35,24 @@
  *  @since 4.5.0
  */
 @Singleton
+@Secondary
 public class ResourceRetrieverJwksClient implements JwksClient {
 
     private static final Logger LOG = LoggerFactory.getLogger(ResourceRetrieverJwksClient.class);
 
     private final DefaultResourceRetriever resourceRetriever = new DefaultResourceRetriever(0, 0, 0);
 
     @Override
-    public String load(String providerName, String url) {
+    @SingleResult
+    public Publisher<String> load(String providerName, String url) {
         try {
             Resource resource = resourceRetriever.retrieveResource(new URL(url));
-            return resource.getContent();
+            return Mono.just(resource.getContent());
         } catch (IOException e) {
             if (LOG.isErrorEnabled()) {
                 LOG.error("Exception loading JWK from " + url, e);
             }
+            return Mono.empty();
         }
-        return null;
```

Comment:
`resourceRetriever.retrieveResource` is a blocking call, so I think it should be wrapped in `Mono.fromCallable`:
```suggestion
        return Mono.fromCallable(() -> {
            try {
                return resourceRetriever.retrieveResource(new URL(url));
            } catch (IOException e) {
                if (LOG.isErrorEnabled()) {
                    LOG.error("Exception loading JWK from " + url, e);
                }
                return null;
            }
        }).map(Resource::getContent).subscribeOn(Schedulers.boundedElastic());
```

Reply:
resolved by https://github.com/micronaut-projects/micronaut-security/pull/1523/commits/3dd723c14d830adfa946b70c7c8f741a5c673724